apps/emqx/src/emqx_packet.erl

```diff
@@ -616,9 +614,14 @@ format_password(undefined) -> "";
 format_password(<<>>) -> "";
 format_password(_Password) -> "******".
 
-format_payload(Payload, text) -> ["Payload=", io_lib:format("~ts", [Payload])];
-format_payload(Payload, hex) -> ["Payload(hex)=", encode_hex(Payload)];
-format_payload(_, hidden) -> "Payload=******".
+format_payload(Payload, text) when ?MAX_PAYLOAD_FORMAT_LIMIT(Payload) ->
+    ["Payload=", unicode:characters_to_list(Payload)];
+format_payload(Payload, hex) when ?MAX_PAYLOAD_FORMAT_LIMIT(Payload) ->
+    ["Payload(hex)=", binary:encode_hex(Payload)];
+format_payload(_, hidden) ->
+    "Payload=******";
+format_payload(Payload, _) ->
+    ["Payload=Large data detected (", integer_to_list(byte_size(Payload)), " bytes)"].
```

Comment:
Agree with @lafirest, perhaps it's better to display `?MAX_PAYLOAD_FORMAT_SIZE` characters in this case